### DISCUSSION TITLE
[A-L] (2023/24) Foglio 1 - Esercizio 2

### DISCUSSION THREAD

**Discussion by Elia-Belli** [sapienzastudentsnetwork/algebra#157] [Esercizi - Viaggi/Piazza]
![es 2](https://github.com/Elia-Belli/algebra2324/assets/107944476/ab736bfc-f901-4aae-b7c2-f14f17790886)

**Comment by Frisbon** [comment]
**Esercizio fatto quando ancora c'erano le lezioni, potrebbe essere sbagliato, infatti ho calcolato $R_1 \circ R_2$ invece che $R_2 \circ R_1 $** 

![image](https://github.com/sapienzastudentsnetwork/algebra2324/assets/84731341/29f6de91-6571-4632-8d5c-ba62a584e55a)

Grazie @CuriousCI per la correzione/chiarimento

**Reply by IonutCicio** [reply]
Attento per la composizione! Quando c'è scritto $R_2 \circ R_1$ devi prima partire da $R_1$ e poi andare in $R_2$.

In $R_1$ hai $(a, a)$ in $R_2$ hai $(a, d)$ quindi in $R_2 \circ R_1$ hai $(a, d)$.

**Reply by FeddyLix17** [reply]
non è invece giusto come ha fatto lui considerando che $R_1 \circ R_2 \neq R_2 \circ R_1$?

**Reply by IonutCicio** [reply]
Appunto $R_1 \circ R_2 \neq R_2 \circ R_1$, il problema è che lui ha calcolato $R_1 \circ R_2$ ma l'esercizio chiedeva $R_2 \circ R_1$. Quando calcoli la composizione la devi fare al contrario rispetto a come la scrivi. $(R_2 \circ R_1)(x) = R_2(R_1(x))$ quindi prima calcoli l'immagine di $x$ rispetto a $R_1$, poi passi l'immagine a $R_2$.

> Mi scuso per l'abuso pazzesco di notazione

**Reply by FeddyLix17** [reply]
Quindi calcolare le composizioni in questo modo

$$
    \Large R_1 \circ R_2 = \lbrace (x,z)\ |\ \exists y \in A\ :\ x\ R_1\ y\ \wedge\ y\ R_2\ z \rbrace
$$

$$
    \Large R_2 \circ R_1 = \lbrace (x,z)\ |\ \exists y \in A\ :\ x\ R_2\ y\ \wedge\ y\ R_1\ z \rbrace
$$

risulta sbagliato poichè fatto "al contrario"?

**Reply by IonutCicio** [reply]
esatto

**Reply by FeddyLix17** [reply]
Buono a sapersi🥹, grazie!

**Comment by FeddyLix17** [comment]
soluzione del punto 3 aggiornata dopo un confronto con @CuriousCI 

1.  $\Large R_1 \cup R_2$

$$
    \Large R_1 \cup R_2 =  \lbrace (x,y)\ |\ (x,y) \in R_1\ \vee\ (x,y) \in R_2 \rbrace = 
$$

$$
    \Large = \lbrace (a,a),\ (a,c),\ (a,d),\ (a,e),\ (b,c),\ (b,d),\ (d,d),\ (d,e) \rbrace
$$

2. $\Large R_1 \cap R_2$

$$
    \Large R_1 \cap R_2 = \lbrace (x,y)\ |\ (x,y) \in R_1\ \wedge\ (x,y) \in R_2 \rbrace =  \lbrace (b,c) \rbrace 
$$

3. $\Large R_2 \circ R_1$

$$
    \Large R_2 \circ R_1 = \lbrace (x,z)\ |\ \exists y \in A\ :\ x\ R_1\ y\ \wedge\ y\ R_2\ z \rbrace = \lbrace (a,d) \rbrace 
$$
